python/ray/data/_internal/execution/resource_manager.py

```diff
@@ -176,6 +179,12 @@ def update_usages(self):
             # DatasetStats and updated on the Ray Data dashboard.
             op._metrics.obj_store_mem_used = op_usage.object_store_memory
 
+            # Update the per node metrics
+            if DataContext.get_current().enable_per_node_metrics:
+                memory_usage_per_node = estimate_object_store_memory_per_node(op, state)
+                for node_id, usage in memory_usage_per_node.items():
+                    op._metrics._per_node_metrics[node_id].obj_store_mem_used = usage
+
```

Comment:
Let's abstract this to a method inside Metrics class itself (include L180 in it as well)